lib/Conversion/TorchOnnxToTorch/ComMicrosoftDomain.cpp

```diff
@@ -63,4 +63,393 @@ void mlir::torch::onnx_c::populateComMicrosoftDomain(
+              loc, positionIdsType, positionIdsB, pastSeqLens,
+              /*alpha=*/cstIntOne);
+
+          totalSeqLens = rewriter.create<Torch::AtenViewOp>(
+              loc, seqLensViewType, totalSeqLens, viewSizeList);
+          Value cond = rewriter.create<Torch::AtenLtTensorOp>(
+              loc,
+              positionIdsType.getWithSizesAndDtype(positionIdsType.getSizes(),
+                                                   rewriter.getI1Type()),
+              positionIdsB, totalSeqLens);
+
+          Value cstOneTensorDataList =
+              rewriter.create<Torch::PrimListConstructOp>(
+                  loc,
+                  rewriter.getType<Torch::ListType>(
+                      rewriter.getType<Torch::IntType>()),
+                  SmallVector<Value>{cstIntOne});
+          Value cstOneTensor = rewriter.create<Torch::AtenTensorOp>(
+              loc,
+              Torch::ValueTensorType::get(
+                  context, {},
+                  IntegerType::get(context, 64, IntegerType::Signed)),
+              cstOneTensorDataList, /*dtype=*/cstInt64Dtype,
+              /*layout=*/cstNone, /*requires_grad=*/cstFalse);
+
+          positionIdsB = rewriter.create<Torch::AtenWhereSelfOp>(
+              loc, positionIdsType, cond, positionIdsB, cstOneTensor);
+
+          isSubsequentPrompt = rewriter.create<Torch::AtenIntBoolOp>(
+              loc, rewriter.getType<Torch::IntType>(), isSubsequentPrompt);
+          isSubsequentPrompt = rewriter.create<Torch::AtenFullOp>(
+              loc,
+              Torch::ValueTensorType::get(context, positionIdsSizeInt,
+                                          rewriter.getI1Type()),
+              /*size=*/posIdsSizeList, /*fill_value=*/isSubsequentPrompt,
+              /*dtype=*/
+              rewriter.create<Torch::ConstantIntOp>(
+                  binder.getLoc(), rewriter.getI64IntegerAttr(
+                                       (int)torch_upstream::ScalarType::Bool)),
+              /*layout=*/cstNone, /*device=*/cstNone, /*pin_memory=*/cstNone);
+          Value positionIds = rewriter.create<Torch::AtenWhereSelfOp>(
+              loc, positionIdsType, isSubsequentPrompt, positionIdsB,
+              positionIdsA);
+
+          qRotary = rewriter.create<Torch::OnnxVariantRotaryEmbeddingOp>(
+              loc, qInput.getType(), qInput, positionIds, cosCache, sinCache,
+              cstInterleaved, /*is_packed_batching=*/cstIntZero,
+              /*num_heads=*/cstIntZero, /*rotary_embedding_dim=*/cstIntZero,
+              /*scale=*/cstFloatOne);
+
+          kRotary = rewriter.create<Torch::OnnxVariantRotaryEmbeddingOp>(
+              loc, qInput.getType(), kInput, positionIds, cosCache, sinCache,
+              cstInterleaved, /*is_packed_batching=*/cstIntZero,
+              /*num_heads=*/cstIntZero, /*rotary_embedding_dim=*/cstIntZero,
+              /*scale=*/cstFloatOne);
+        }
+
+        // Do attention.
```

Comment:
// Compute the attention score and apply the score to V

Reply:
That comment is present there since it does the attention computation in 2 steps but here we're not doing the same.